### PR TITLE
Integrate rules list

### PR DESCRIPTION
**Pull request by sans-harness**
- Adds the `repo-branch-rules-page` to `gitness` with minor fixes. 
- Uses HTTP restful URL pattern, no query params. 
- changes button to `update rule` on rule edit. 
- Minor fixes such as increasing pipe size, text no longer breaking, icons made smaller, etc.
- Adds `dropdown` in input to `include/exclude` button. 

e2e vid:

https://github.com/user-attachments/assets/e9218506-17a8-4b13-b90b-040f60a86580

### CODE REVIEW

File: apps/gitness/src/App.tsx [hunk 1]
```diff
@@ -370,7 +370,13 @@ export default function App() {
                     },
                     {
                       path: 'rules',
-                      element: <RepoBranchSettingsRulesPageContainer />
+                      element: <RepoBranchSettingsRulesPageContainer />,
+                      children: [
+                        {
+                          path: ':identifier',
+                          element: <RepoBranchSettingsRulesPageContainer />
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/App.tsx:377]: why is this element same as the `rules` page element above? shouldn't we be rendering the "CreateRulePage" or similar?

File: apps/gitness/src/pages/repo-sandbox/repo-sandbox-branch-rules-container.tsx [hunk 2]
```diff
@@ -1,82 +1,53 @@
...
     mutate: addRule,
     error: addRuleError,
-    isSuccess: addRuleSuccess,
     isLoading: addingRule
-  } = useRuleAddMutation({ repo_ref: repoRef })
+  } = useRuleAddMutation(
+    { repo_ref: repoRef },
+    {
+      onSuccess: () => {
+        const repoName = repoRef.split('/')[1]
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/pages/repo-sandbox/repo-sandbox-branch-rules-container.tsx:42]: can we change the `useGetRepoRef` function to return both the repo ref and the extracted repoName as well? this seems like a re-usable piece of code.

File: apps/gitness/src/utils/repo-branch-rules-utils.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  BranchRuleId.REQUIRE_NO_CHANGE_REQUEST,
+  BranchRuleId.COMMENTS,
+  BranchRuleId.STATUS_CHECKS,
+  BranchRuleId.MERGE,
+  BranchRuleId.DELETE_BRANCH
+]
+
+// Util to transform API response into expected-form format for branch-rules-edit
+
+const extractBranchRules = (data: RuleGetOkResponse) => {
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/utils/repo-branch-rules-utils.ts:27]: can we add a return type on this?

File: apps/gitness/src/utils/repo-branch-rules-utils.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+      checked,
+      submenu,
+      selectOptions
+    })
+  }
+
+  return rules
+}
+
+export const transformDataFromApi = (data: RuleGetOkResponse) => {
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/utils/repo-branch-rules-utils.ts:72]: same here.. let's add strong return types for all utils

File: packages/playground/src/components/repo-settings/repo-settings-general/repo-settings-general-rules.tsx [hunk 5]
```diff
@@ -1,7 +1,8 @@
 import React from 'react'
-import { Button, ListActions, SearchBox, Icon, Text, StackedList } from '@harnessio/canary'
+import { RuleDataType, ErrorTypes } from './types'
+import { Button, ListActions, SearchBox, Icon, Text, StackedList, Spacer } from '@harnessio/canary'
 
-const Title = ({ title, iconName }: { title: string; iconName: 'green-tick' | 'cancel-grey' }) => {
+const Title = ({ title, iconName }: { title: string | undefined; iconName: 'green-tick' | 'cancel-grey' }) => {
```
Comment (abhinavrastogi-harness) [inline packages/playground/src/components/repo-settings/repo-settings-general/repo-settings-general-rules.tsx:5]: `title?: string` might be a more readable type

**Review by abhinavrastogi-harness — APPROVED** [review]

File: apps/gitness/src/App.tsx (hunk 1 above)
Comment (sans-harness) [reply apps/gitness/src/App.tsx:377]: Not really, its the same form component...which on edit is rendered with prefilled values.

File: apps/gitness/src/pages/repo-sandbox/repo-sandbox-branch-rules-container.tsx (hunk 2 above)
Comment (sans-harness) [reply apps/gitness/src/pages/repo-sandbox/repo-sandbox-branch-rules-container.tsx:42]: Sure.

**Pull request merged**